crates/agglayer-rpc/src/lib.rs

```
@@ -347,6 +348,57 @@ where
         }
     }
 
+    pub fn get_last_settled_claim(
+        &self,
+        network_id: NetworkId,
+        height: Height,
+    ) -> Result<Option<SettledClaim>, GetLatestSettledClaimError> {
+        // Iterate from the given height down to 0
+        for current_height in (0..=height.as_u64()).rev().map(Height::from) {
+            // Fetch certificate header for the current height
+            let header_opt = self
+                .state
+                .get_certificate_header_by_cursor(network_id, current_height)?;
+
+            let header = match header_opt {
+                Some(h) => h,
+                None => continue, // No certificate at this height, continue to next lower height
+            };
+
+            // Only proceed if both epoch_number and certificate_index are present
+            let (epoch_number, certificate_index) =
+                match (header.epoch_number, header.certificate_index) {
+                    (Some(epoch), Some(idx)) => (epoch, idx),
+                    _ => continue,
+                };
+
+            // Fetch the certificate from the epoch store
+            let certificate_opt = self
+                .epochs_store
+                .get_certificate(epoch_number, certificate_index)
+                .map_err(GetLatestSettledClaimError::from)?;
+
+            let certificate = match certificate_opt {
+                Some(cert) => cert,
+                None => continue,
```

Same here, should we return an error?